metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/S3Dataset.java

```diff
@@ -0,0 +1,118 @@
+package datahub.spark.model.dataset;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.linkedin.common.FabricType;
+import com.typesafe.config.Config;
+
+import datahub.spark.SparkConfigUtil;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class S3Dataset extends SparkDataset {
+
+  private static final String TABLE = "{table}";
+  private static final String TABLE_MARKER = "/{table}";
+  private static final String TABLE_MARKER_REGEX = "/\\{table\\}";
+
+  public S3Dataset(String pathUri, String platformInstance, FabricType fabricType) {
+    super("s3", platformInstance, pathUri, fabricType);
+  }
+
+  public S3Dataset(String pathUri) {
+    super("s3", pathUri);
+  }
+
+  public static boolean isMatch(String pathUri) {
+    return isS3Uri(pathUri);
+  }
+
+  public static S3Dataset create(String pathUri) throws InstantiationException {
+    if (!isMatch(pathUri)) {
+      throw new InstantiationException("S3Dataset instance cannot be created for non S3 pathUri: " + pathUri);
+    }
+    List<String> pathAliasList = SparkConfigUtil.getS3PathAliasList();
+    log.debug("path_alias_list: " + String.join(",", pathAliasList));
+    if (pathAliasList != null && pathAliasList.size() > 0) {
+      for (String pathAlias : SparkConfigUtil.getS3PathAliasList()) {
+        log.debug("Checking match for path_alias: " + pathAlias);
+        Config pathSpecConfig = SparkConfigUtil.getS3PathAliasDetails(pathAlias);
+        if (pathSpecConfig != null) {
+          String s3Uri = getS3Uri(pathUri, SparkConfigUtil.getPathSpecList(pathSpecConfig));
+          if (s3Uri != null) {
+            String platformInstance = SparkConfigUtil.getPlatformInstance(pathSpecConfig);
+            FabricType fabricType = SparkConfigUtil.getFabricType(pathSpecConfig);
+            return new S3Dataset(getDatasetName(s3Uri), platformInstance, fabricType);
+          }
+        }
+
+      }
+      log.debug("nothing matched from path_alias_list. Falling back using config 'path_spec_list'");
+    }
+    String s3Uri = getS3Uri(pathUri, SparkConfigUtil.getS3PathSpecList());
+    if (s3Uri == null) {
```

Comment:
Why can't getS3Uri itself return the pathUri, it's private method, and will reduce this null check, simplify code?